File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int sub = 0 ; 
        for(int i = 0 ; i < s.length() ; i++ ){
            bool charSeen[128] = {false} ;
            int l = 0 ; 
            for(int j = i ; j < s.length() ; j++ ){
                if(charSeen[s[j]]){
                    break ; 
                }
                charSeen[s[j]] = true;
                l++ ; 
            }
            if(l>sub){
                sub = l ;
            }
            l = 0  ; 
        }
        return sub ; 
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        char[] st = s.toCharArray() ; 
        int ma = 0 ;
        for(int i = 0 ; i < st.length ; i++){
            boolean[] ch = new boolean[128] ; 
            int le = 0 ;
            for(int j = i ; j < st.length ; j++){
                if(ch[st[j]]){
                    break ; 
                }
                ch[st[j]] = true ; 
                le++ ; 
            }
            ma = Math.max(le, ma) ; 
        }
        return ma ; 
    }
}